src/core_modules/capture-core/components/Pages/New/NewPage.container.js

```diff
@@ -108,6 +109,11 @@ export const NewPage: ComponentType<{||}> = () => {
           || dataEntryHasChanges(state, 'newPageDataEntryId-newTei'),
     );
 
+    const [newPageKey, setNewPageKey] = useState();
+    const handleRefreshNewTeForm = () => {
+        setNewPageKey(uuid());
+    };
```

Comment:
Hi @henrikmv, I believe the issue is that [history.push](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/utils/routing/useNavigate.js#L8) doesn't have any effect when the path is the same (there's a warning in the console). However, when I use `history.replace` along with a useEffect like in [this example](https://jasonwatmore.com/react-router-v6-listen-to-location-route-change-without-history-listen), the form reloads properly. Any thoughts?

<img width="1038" alt="Screenshot 2025-04-28 at 10 59 33" src="https://github.com/user-attachments/assets/528ab088-1e23-47b4-b7d2-35834fcc1136" />

Thanks!